static/js/timerender.js

```diff
@@ -307,21 +307,26 @@ export const absolute_time = (function () {
 })();
 
 export function get_full_datetime(time) {
-    const date_string_options = {weekday: "long", month: "long", day: "numeric"};
-    const time_string_options = {timeStyle: "full"};
+    // Convert to number of hours ahead/behind UTC.
+    // The sign of getTimezoneOffset() is reversed wrt
+    // the conventional meaning of UTC+n / UTC-n
+    const tz_offset = -time.getTimezoneOffset() / 60;
+
+    const options = {dateStyle: "short", timeStyle: "long"};
 
     if (page_params.twenty_four_hour_time) {
-        time_string_options.hourCycle = "h24";
+        options.hourCycle = "h24";
     }
 
-    const current_date = new Date();
-    if (time.getFullYear() !== current_date.getFullYear()) {
-        // Show year only if not current year.
-        date_string_options.year = "numeric";
-    }
+    let time_string = time.toLocaleString(undefined, options);
```

Comment:
Used `undefined` as the `locale` to  use users' browser default locale. The locale setting we can get from `page_params.request_language` is just language but not locale. Using that might give `GMT+5:30` kind of timezone string for different locales. (for eg: `en-IN` works well in Indian timezone but not `en`)